xdem/volume.py

```diff
@@ -283,3 +284,135 @@ def hypsometric_interpolation(voided_ddem: Union[np.ndarray, np.ma.masked_array]
     assert output is not None
 
     return output
+
+
+def local_hypsometric_interpolation(voided_ddem: Union[np.ndarray, np.ma.masked_array],
+                                    ref_dem: Union[np.ndarray, np.ma.masked_array],
+                                    mask: np.ndarray, min_coverage: float = 0.2,
+                                    plot: bool = False) -> np.ma.masked_array:
+    """
+    Interpolate a dDEM using local hypsometric interpolation.
+    The algorithm loops through each features in the vector file.
+
+    The dDEM is assumed to have been created as "voided_ddem = reference_dem - other_dem".
+
+    :param voided_ddem: A dDEM with voids (either an array with nans or a masked array).
+    :param ref_dem: The reference DEM in the dDEM comparison.
+    :param mask: A raster of same shape as voided_ddem and ref_dem, containing a diferent non-0 pixel value for \
+each geometry on which to loop.
+    :param min_coverage: The minimum coverage fraction to be considered for interpolation. Default is 0.2.
+    :param plot: Set to True to display intermediate plots.
```

Comment:
Could you add a `:returns:` line? Would be nice to explain what it returns. Something along:

```
:returns: A dDEM which is interpolated within the mask bounds for all areas filling the min_coverage criterion. 
```
or something more concise.